### ISSUE TITLE
Live-Codeblock  does not support Optional Chaining, But react-live does

### ISSUE THREAD

**Issue by childrentime** [facebook/docusaurus#9809]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [ ] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

"When using the @docusaurus/theme-live-codeblock plugin, an error will be thrown if optional chaining syntax like '123'?.toString() is used."

### Steps to reproduce

1.  use the code live editor https://docusaurus.io/zh-CN/docs/markdown-features/code-blocks#interactive-code-editor
2. add a demo like 
```tsx live
```tsx live
function demo() {
  console.log('????','dasdsa'?.toString())

  return (
    <div>
      <div>hello world</div>
    </div>
  );
};
``
```

### Expected behavior

no error throw 

### Actual behavior

<img width="700" alt="image" src="https://github.com/facebook/docusaurus/assets/58261676/d05c6ab1-3d99-4c23-bbe8-37ec7b6b19e9">

### Your environment

- Docusaurus version used: all latest
- Environment name and version (e.g. Chrome 89, Node.js 16.4): chrome latest node20
- Operating system and version (e.g. Ubuntu 20.04.2 LTS): macos 14.1

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by slorber** [comment]
> But react-live does

See https://commerce.nearform.com/open-source/react-live/#inline-demo

No react-live does not support that

![CleanShot 2024-02-01 at 12 36 10](https://github.com/facebook/docusaurus/assets/749374/5e0f1f38-c138-4c98-b32a-4e90cca19af9)

And Docuaurus shows the exact same error react-live gives us:

![CleanShot 2024-02-01 at 12 36 44](https://github.com/facebook/docusaurus/assets/749374/5ea3ce2b-c24d-4c45-86c4-dd9dad3a0dc2)

---

For this reason I'm closing as this works as intended. 

If you want this to be supported, you should ask on the react-live repo directly: https://github.com/FormidableLabs/react-live

**Comment by childrentime** [comment]
@slorber hi, react-live4.1.6 fixed this. Turns out that It is a bug of react-live.

**Comment by csantos-nydig** [comment]
There is something odd in react-live. from what I can see this is still not working in `react-live@4.1.6`.

@childrentime how did `react-live@4.1.6` fix this issue for you? I don't see anything in their diff that would indicate a fix for this issue.

I logged an issue in their side this morning: https://github.com/FormidableLabs/react-live/issues/389

As of now:

- 🛑 it doesn't work in https://docusaurus.io/docs/markdown-features/code-blocks#interactive-code-editor
- 🛑 it doesn't work in [codesandbox](https://codesandbox.io/p/devbox/prod-monad-7gh39v?file=%2Fdocusaurus.config.ts) using `docusaurus@3.4.0`
- 🛑 it doesn't work in [codesandbox](https://codesandbox.io/p/devbox/react-live-optional-chaining-q2qwy2?file=%2Fsrc%2FApp.tsx%3A4%2C1-17%2C2) using `react-live@4.1.6` alone
- 😕 it works in https://commerce.nearform.com/open-source/react-live

I'd appreciate if docusaurus team (@slorber, @Josh-Cena) could add support to the react-live issue because it's not longer working as intended 🙏

Little additional note: this used to work **before** `react-live@4.0.0` ([see codesandbox](https://codesandbox.io/p/devbox/react-live-v3-optional-chaining-5mnh4f?file=%2Fpackage.json%3A12%2C20)), hence it used to work in docusaurus v2, so:

- it's a bug they introduced in `v4.0.0`
- it's a bug docusaurus introduced when upgrading react-live

**Comment by childrentime** [comment]
Interesting. I fixed this bug before actually

**Comment by childrentime** [comment]
works in my website: https://www.reactuse.com/element/useActiveElement

**Comment by childrentime** [comment]
https://github.com/childrentime/reactuse You can refer to this repo. Sorry for tap so much mesaggs cause it is late in my timezone and i am in bed.

**Comment by csantos-nydig** [comment]
@childrentime let's move this conversation over to https://github.com/FormidableLabs/react-live/issues/389 to keep docusaurus!

**Comment by slorber** [comment]
As far as I understand, things have been resolved in `react-live@4.1.7`:

https://github.com/FormidableLabs/react-live/issues/389#issuecomment-2189769666
